horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -58,32 +64,403 @@ public class RunServiceTest extends BaseServiceTest {
+
+      assertNull(dataSetQueue.poll(50, TimeUnit.MILLISECONDS));
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testTransformationWithoutMatchFirstLevel(TestInfo info) throws InterruptedException {
+      Schema schema = createExampleSchema("Aca", "Aca", "Aca", false);
+      testTransformationWithoutMatch(info, schema, runWithValue(schema, 42));
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testTransformationWithoutMatchSecondLevel(TestInfo info) throws InterruptedException {
+      Schema schema = createExampleSchema("B", "B", "B", false);
+      testTransformationWithoutMatch(info, schema, JsonNodeFactory.instance.objectNode().set("nested", runWithValue(schema, 42)));
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testMultiSchemaWithoutTransformer(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+
+      Schema schemaA = createExampleSchema("AAAA", "AAAAInc", "AAAARrUs", false);
+      Schema schemaB = createExampleSchema("BBBB", "BBBBInc", "BBBBRrUs", false);
+      Schema schemaC = createExampleSchema("CCCC", "CCCCInc", "CCCCRrUs", false);
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+
+      ObjectNode data = runWithValue(42.0d, schemaA, schemaB, schemaC);
+
+      uploadRun(data,test.name);
+
+      DataSet dataSet = dataSetQueue.poll(POLL_DURATION_SECONDS, TimeUnit.SECONDS);
+      JsonNode eventData = dataSet.data;
+      assertTrue(eventData.isArray());
+      assertEquals(3, eventData.size());
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testSchemaTransformerWithExtractorProducingNullValue(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+
+      Schema schema = createExampleSchema("DDDD", "DDDDInc", "DDDDRrUs", false);
+      NamedJsonPath scalarPath = createExampleNamedJsonPath("sheep", "$.duff");
+      String scalarFunction = createFunction("sheep");
+      Transformer scalarTransformer = createExampleTransformerWithJsonPath("tranProcessNullExtractorValue", schema, scalarFunction, scalarPath);
+
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      addTransformer(test, scalarTransformer);
+
+      ObjectNode data = runWithValue(42.0d, schema);
+
+      uploadRun(data,test.name);
+
+      DataSet dataSet = dataSetQueue.poll(POLL_DURATION_SECONDS, TimeUnit.SECONDS);
+      JsonNode eventData = dataSet.data;
+      assertTrue(eventData.isArray());
+      assertEquals(1, eventData.size());
+      JsonNode sheep = eventData.path(0).path("outcome").path("sheep");
+      assertEquals("null", sheep.asText());
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testOnlyOneTransformerOfTwoMatches(TestInfo info) throws InterruptedException {
```

Comment:
Isn't this the same as `testTransformationChoosingSchema`?

Reply:
It is very similar. It was added because of your comment. I will remove `testOnlyOneTransformerOfTwoMatches`.